### PR TITLE
Add ContractTrust

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-devpack-dotnet/issues/581

### CODE REVIEW

**Comment by erikzhang** [comment]
`trusts` can be an array or `*`. If it is an array, the elements should be the hash of a contract, or the public key of a group.

**Comment by shargon** [comment]
We can add a new attribute for each hash/pubKey, but I will update the code for an array

**Comment by erikzhang** [comment]
I mean we should consider `*`, and verify the elements of the array.

File: src/Neo.Compiler.CSharp/CompilationContext.cs [hunk 1]
```diff
@@ -277,7 +278,7 @@ public JObject CreateManifest()
                     }).ToArray()
                 },
                 ["permissions"] = permissions.ToJson(),
-                ["trusts"] = new JArray(),
+                ["trusts"] = trusts.OrderBy(p => p).Select(u => new JString(u)).ToArray(),
```
Comment (erikzhang) [inline src/Neo.Compiler.CSharp/CompilationContext.cs:281]: It can be an array or `*`. `*` is not allowed to be an element.

Comment (shargon) [reply src/Neo.Compiler.CSharp/CompilationContext.cs]: Done

**Review by erikzhang — APPROVED** [review]

File: src/Neo.Compiler.CSharp/CompilationContext.cs [hunk 2]
```diff
@@ -356,6 +366,12 @@ private void ProcessClass(SemanticModel model, INamedTypeSymbol symbol)
                         case nameof(scfx.Neo.SmartContract.Framework.ContractPermissionAttribute):
                             permissions.Add((string)attribute.ConstructorArguments[0].Value!, attribute.ConstructorArguments[1].Values.Select(p => (string)p.Value!).ToArray());
                             break;
+                        case nameof(scfx.Neo.SmartContract.Framework.ContractTrustAttribute):
+                            string trust = (string)attribute.ConstructorArguments[0].Value!;
+                            if (!ValidateContractTrust(trust))
+                                throw new ArgumentException($"The value {trust} is not a valid one for ContractTrust");
```
Comment (RichardBelsum) [inline src/Neo.Compiler.CSharp/CompilationContext.cs:372]: Hold on, Trusts should only be used for wallet warnings and should not be mandatory.

Whether the contract can be called or not, should depend only on the Permissions and WitnessScope, not Trust.

Comment (roman-khimov) [reply src/Neo.Compiler.CSharp/CompilationContext.cs:372]: Reminds me of neo-project/neo#1664.

Comment (chenzhitong) [reply src/Neo.Compiler.CSharp/CompilationContext.cs:372]: > If a contract is trusted, the user interface will not give any warnings when called by the contract.
https://github.com/neo-project/neo/blob/7e2d435aada96e7e22c7564984bff1419377c33f/src/neo/SmartContract/Manifest/ContractManifest.cs#L51

Changed again?

Comment (erikzhang) [reply src/Neo.Compiler.CSharp/CompilationContext.cs:372]: I think it's a format check.

**Pull request merged**